packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -422,81 +433,79 @@ class _DecoyChild extends StatefulWidget {
+        ).chain(CurveTween(curve: Curves.linear)),
+        weight: 3,
       ),
       TweenSequenceItem<Rect?>(
         tween: RectTween(
-          begin: midRect,
+          begin: widget.beginRect,
           end: widget.endRect,
-        ).chain(CurveTween(curve: Curves.easeOutCubic)),
-        weight: 1.0,
+        ).chain(CurveTween(curve: Curves.easeOutSine)),
+        weight: 5,
       ),
     ]).animate(widget.controller);
-    _rect.addListener(_rectListener);
-  }
 
-  // Listen to the _rect animation and vibrate when it reaches the halfway point
-  // and switches from animating down to up.
-  void _rectListener() {
-    if (widget.controller.value < 0.5) {
-      return;
-    }
-    HapticFeedback.selectionClick();
-    _rect.removeListener(_rectListener);
+    const List<BoxShadow> endBoxShadow = <BoxShadow>[
+      BoxShadow(
+        color: Color(0x40000000),
+        blurRadius: 10.0,
+        spreadRadius: 0.5,
+      )
+    ];
+
+    _boxDecoration = DecorationTween(
+      begin: BoxDecoration(
+          color: const Color(0xFFFFFFFF),
+          borderRadius: widget.child.runtimeType == Container
+              ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                      ?.borderRadius ??
+                  BorderRadius.circular(0)
+              : BorderRadius.circular(0),
+          boxShadow: widget.child.runtimeType == Container
+              ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                      ?.boxShadow ??
+                  const <BoxShadow>[]
+              : const <BoxShadow>[]),
+      end: BoxDecoration(
+        color: const Color(0xFFFFFFFF),
+        borderRadius: widget.child.runtimeType == Container
+            ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                    ?.borderRadius ??
+                BorderRadius.circular(0)
+            : BorderRadius.circular(0),
+        boxShadow: widget.child.runtimeType == Container
+            ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                    ?.boxShadow ??
+                endBoxShadow
+            : endBoxShadow,
+      ),
+    ).animate(widget.controller);
   }
 
   @override
   void dispose() {
-    _rect.removeListener(_rectListener);
     super.dispose();
   }
 
   Widget _buildAnimation(BuildContext context, Widget? child) {
-    final Color color = widget.controller.status == AnimationStatus.reverse
-      ? _masklessColor
-      : _mask.value;
     return Positioned.fromRect(
       rect: _rect.value!,
-      child: ShaderMask(
-        key: _childGlobalKey,
-        shaderCallback: (Rect bounds) {
-          return LinearGradient(
-            begin: Alignment.topLeft,
-            end: Alignment.bottomRight,
-            colors: <Color>[color, color],
-          ).createShader(bounds);
-        },
-        child: widget.child,
-      ),
+        child:
+            Container(decoration: _boxDecoration.value, child: widget.child)
```

Comment:
Formatting: Put the Container on the first line and split the parameters onto new lines.

Reply:
Also trailing comma.